File: Python/Hash_Sort.py
def hash_sort(arr):
    hash_table = {}
    max_val = 0

    for num in arr:
        hash_table[num] = hash_table.get(num, 0) + 1
        if num > max_val:
            max_val = num

    sorted_arr = []
    for i in range(max_val + 1):
        if i in hash_table:
            sorted_arr.extend([i] * hash_table[i])
    return sorted_arr

arr = [4, 2, 2, 8, 3, 3, 1]
print("Arreglo original:", arr)

arr_sorted = hash_sort(arr)
print("Arreglo ordenado:", arr_sorted)

File: JavaScript/Hash_Sort.js
function hashSort(arr) {
    let hashTable = {};
    let maxVal = 0;

    for (let num of arr) {
        hashTable[num] = (hashTable[num] || 0) + 1;
        if (num > maxVal) maxVal = num;
    }

    let sortedArr = [];
    for (let i = 0; i <= maxVal; i++) {
        if (hashTable[i]) {
            for (let j = 0; j < hashTable[i]; j++) {
                sortedArr.push(i);
            }
        }
    }
    return sortedArr;
}

let arr = [4, 2, 2, 8, 3, 3, 1];
console.log("Arreglo original:", arr);

let sortedArr = hashSort(arr);
console.log("Arreglo ordenado:", sortedArr);